app/models/manageiq/providers/cisco_intersight/inventory/parser/physical_infra_manager.rb

```diff
@@ -3,200 +3,472 @@ class Inventory::Parser::PhysicalInfraManager < Inventory::Parser
 
     def parse
       physical_servers
-      physical_server_details
       physical_racks
-      hardwares
-      firmwares
       physical_chassis
-      physical_chassis_details
+      physical_switches
     end
 
-    def physical_servers
-      collector.physical_servers.each do |s|
-        registered_device_moid = get_registered_device_moid(s)
-        device_registration = collector.get_asset_device_registration_by_moid(registered_device_moid)
-        # Setting value of chassis and rack to nil for now
-        chassis = nil # TODO: After chassis gets written into the DB, obtain it and write it here as reference using lazy find.
-        rack = nil # TODO: After chassis gets written into the DB, obtain it and write it here as reference using lazy find.
-        server = persister.physical_servers.build(
-          :ems_ref          => s.moid,
-          :health_state     => get_health_state(s), # health_state obtained through atribute s.alarm_summary
-          :hostname         => device_registration.device_hostname[0], # I assume one registered device manages one (and only one) compute element
-          :name             => s.name,
-          :physical_chassis => chassis, # nil for now
-          :physical_rack    => rack, # nil for now
-          :power_state      => s.oper_power_state,
-          :raw_power_state  => s.oper_power_state,
-          :type => "ManageIQ::Providers::CiscoIntersight::PhysicalInfraManager::PhysicalServer",
-        )
-        persister.physical_server_computer_systems.build(
-          :managed_entity => server
-        )
-      end
-    end
+    # Methods that are executed during refresh call
 
-    def physical_server_details
-      collector.physical_servers.each do |s|
-        registered_device_moid = get_registered_device_moid(s)
-        server = persister.physical_servers.lazy_find(s.moid)
-        source_object = collector.get_source_object_from_physical_server(s)
-        device_contract_information_unit = collector.get_device_contract_information_from_device_moid(registered_device_moid)
-        persister.physical_server_details.build(
-          :description        => device_contract_information_unit.product.description,
-          :location           => format_location(device_contract_information_unit),
-          :location_led_state => get_locator_led_state(source_object),
-          :machine_type       => device_contract_information_unit.device_type,
-          :model              => s.model,
-          :product_name       => s.name,
-          :resource           => server,
-          :room               => s.slot_id,
-          :serial_number      => s.serial,
-        )
-      end
-    end
+    def physical_servers
+      # Parses physical servers and its details
+      collector.physical_servers.each do |server|
+        # build collection physical_servers
+        physical_server = build_physical_servers(server)
```

Comment:
Should be singular, we're only building one physical server here
```suggestion
        physical_server = build_physical_server(server)
```